qa/websocket_server_fee_payments.py

```diff
@@ -0,0 +1,252 @@
+        assert_true('feePayments' in event_payload)
+        assert_equal(expected_fee_payments, event_payload['feePayments']['newBoxes'],
+                     "Different fee payments retrieved for the block.")
+
+        # Send get single block request with block height of the last block of the WE
+        response = json.loads(ws.sendMessage(ws_connection,
+                               ws.REQUEST_MSG_TYPE,
+                               0,
+                               ws.GET_SINGLE_BLOCK_REQUEST,
+                               json.dumps({"height": sc_last_we_block_height})))
+        # Verify static response field
+        ws.checkMessageStaticFields(response,
+                               ws.RESPONSE_MSG_TYPE,
+                               0,
+                               ws.GET_SINGLE_BLOCK_REQUEST)
+        # Verify responsePayload
+        response_payload = response['responsePayload']
+        assert_equal(response_payload['hash'], sc_last_we_block_id)
+        assert_equal(response_payload['height'], sc_last_we_block_height)
+
+        assert_true('feePayments' in event_payload)
+        assert_equal(expected_fee_payments, event_payload['feePayments']['newBoxes'],
+                     "Different fee payments retrieved for the block.")
+
+        ###########################################################
+        #       Check new tip event without payments info         #
+        ###########################################################
+        print("######## Check new tip event without payments info test ########")
+
+        # Generate 1 SC block that should not cause to any fee payments
+        self.blocks.append(generate_next_blocks(sc_node, "", 1)[0])
+
+        # Check the event type and get the response
+        for i in range(0, 2):
+            response = json.loads(ws_connection.recv())
+            if response['msgType'] == 0 and response['answerType'] == 0:  # new Tip event
+                new_tip_event = response
+            elif response['msgType'] == 0 and response['answerType'] == 2:  # changed mempool event
+                pass  # skip mempool event
+            else:
+                raise Exception("Message not expected")  # We shouldn't receive any other message
+
+        # After generating a block we expect to receive new tip event
+        # Verify static response field
+        ws.checkMessageStaticFields(new_tip_event,
+                                    ws.EVENT_MSG_TYPE,
+                                    -1,
+                                    ws.UPDATE_TIP_EVENT)
+        event_payload = new_tip_event['eventPayload']
+
+        assert_equal(event_payload['hash'], self.blocks[-1])
+        assert_equal(event_payload['height'], sc_last_we_block_height + 1)
+        assert_true('feePayments' not in event_payload)
+        expected_fee_payments = http_block_getFeePayments(sc_node, self.blocks[-1])['feePayments']
```

Comment:
Consider to rename expected_fee_payments to api_fee_payments.

Reply:
done